jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jsonb/JsonbDiagnosticsCollector.java

```diff
@@ -147,4 +144,17 @@ private boolean hasJsonbAnnotationOtherThanTransient(IAnnotatable annotable) thr
         }
         return false;
     }
+    
+    private List<String> getJsonbAnnotationNamesForField(IField field) throws JavaModelException {
+        IAnnotation fieldAnnotations[] = field.getAnnotations();
+        List<IAnnotation> fieldAnnotationsList = Arrays.asList(fieldAnnotations);
+        List<String> annotationNames = fieldAnnotationsList.stream().map(
+                fieldAnnotation -> fieldAnnotation.getElementName()).collect(Collectors.toList());
+
+        List<String> jsonbAnnotationNames = new ArrayList<String>();
+        for (String annotation : annotationNames)
+            if (annotation.startsWith(JsonbConstants.JSONB_PREFIX))
```

Comment:
Kathryn and I discussed the definition of Jsonb annotations [here](https://github.com/eclipse/lsp4jakarta/pull/270#discussion_r830281474). It might not always be the case that an annotation starting with JSONB_PREFIX is a Jsonb annotation. I've added a list of all Jsonb annotations in the constant class. We can check whether that list contains a given annotation instead.

Reply:
Thanks for catching that. Done